client/src/pages/AdminPage/ManageSuggestionsPage/ManageSuggestionsPage.js

```diff
@@ -0,0 +1,114 @@
+    }
+
+    componentDidUpdate(prevProps, prevState, snapshot) {
+        const { dispatch, match: { params: { suggestionId } } } = this.props;
+
+        if (prevProps.approveBulkCreateSuggestion.pending && !this.props.approveBulkCreateSuggestion.pending &&
+            Array.isArray(this.props.approveBulkCreateSuggestion.result)) {
+            dispatch(fetchBulkCreateSuggestion(suggestionId));
+        }
+    }
+
+    dispatchSuggestionActions(actions) {
+        const { dispatch, match: { params: { suggestionId } }, location: { search } } = this.props;
+        const type = QueryString.parse(search).type;
+
+        if (suggestionId && type) {
+            try {
+                if (!isNaN(parseInt(suggestionId))) {
+                    switch (type) {
+                        case 'create':
+                            dispatch(actions.create(suggestionId));
+                            break;
+                        case 'update':
+                            dispatch(actions.update(suggestionId));
+                            break;
+                        case 'bulk':
+                            dispatch(actions.bulk(suggestionId));
+                            break;
+                        default:
+                            return;
+                    }
+                }
+            } catch (err) {}
+        }
+    }
+
+    fetchSuggestionIfIdAndTypeExist() {
+        this.dispatchSuggestionActions({create: fetchCreateSuggestion, update: fetchUpdateSuggestion, bulk: fetchBulkCreateSuggestion});
+    }
+
+    handleSuggestionApproveButtonClick() {
+        this.dispatchSuggestionActions({create: approveCreateSuggestion, update: approveUpdateSuggestion, bulk: approveBulkCreateSuggestion});
+    }
+
+    handleSuggestionRejectButtonClick() {
+        this.dispatchSuggestionActions({create: rejectCreateSuggestion, update: rejectUpdateSuggestion, bulk: rejectBulkCreateSuggestion});
+    }
+
+    render() {
+        const { mode, fetchCreateSuggestion, fetchUpdateSuggestion, fetchBulkCreateSuggestion,
+            approveCreateSuggestion, rejectCreateSuggestion, approveUpdateSuggestion, rejectUpdateSuggestion,
+            location: { search }, rejectBulkCreateSuggestion, approveBulkCreateSuggestion } = this.props;
+        const type = QueryString.parse(search).type;
+
+        const suggestion = approveCreateSuggestion.result || rejectCreateSuggestion.result ||
+            approveUpdateSuggestion.result || rejectUpdateSuggestion.result || rejectBulkCreateSuggestion.result ||
+            fetchCreateSuggestion.result || fetchUpdateSuggestion.result || fetchBulkCreateSuggestion.result;
+
+        const pending = approveCreateSuggestion.pending || rejectCreateSuggestion.pending ||
+            approveUpdateSuggestion.pending || rejectUpdateSuggestion.pending || rejectBulkCreateSuggestion.pending ||
+            fetchCreateSuggestion.pending || fetchUpdateSuggestion.pending || fetchBulkCreateSuggestion.pending;
```

Comment:
![image](https://user-images.githubusercontent.com/16584007/79133729-82abca00-7d7a-11ea-9371-999e32e7593a.png)

Reply:
I guess I don't fully understand what the original is doing. Is it that they are all always falsey except for one, so the defined one is always returned? in that case you could do `reducers.find(reducer => reducer.result);`

Reply:
for `pending` I do think `reducers.filter(reducer => reducer.result).length > 0` is correct - you're trying to see if at least one is pending, no?

Reply:
For your first comment: kind of. That's true until you approve a Suggestion. Then `approveCreateSuggestion.result` and `fetchCreateSuggestion.result` both are truthy, but we want `approveCreateSuggestion.result` since it's the updated one hence the specific ordering.
If `find` always traverses the list in order than that won't be an issue.

Reply:
Yeah, you're right for `pending`. I'll fix that

Reply:
Ok, that's a little complex logic there hidden by the simple `||`'s. I would leave a comment explaining that and leave them as is.

Reply:
Good point. I wasn't sure if there was a better way to do it.